app/src/main/java/se/ju/student/robomow/adapter/MowSessionAdapter.kt

```diff
@@ -0,0 +1,89 @@
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import se.ju.student.robomow.R
+import se.ju.student.robomow.model.MowSession
+import se.ju.student.robomow.model.TimeStamp
+import java.time.Instant
+import java.time.LocalDateTime
+import java.time.ZoneOffset
+import java.time.format.DateTimeFormatter
+
+class MowSessionAdapter(
+    private val context: Context,
+    private var mowSessions: List<MowSession>,
+    private val onItemClicked: (MowSession) -> Unit
+) : RecyclerView.Adapter<MowSessionAdapter.MowSessionViewHolder>() {
+    class MowSessionViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
+        val date: TextView = itemView.findViewById(R.id.mow_session_date)
+        val status: TextView = itemView.findViewById(R.id.mow_session_status)
+        val collisions: TextView = itemView.findViewById(R.id.mow_session_collisions)
+        val mowingTime: TextView = itemView.findViewById(R.id.mow_session_mowing_time)
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MowSessionViewHolder {
+        val itemView =
+            LayoutInflater.from(parent.context).inflate(R.layout.mow_session_item, parent, false)
+        return MowSessionViewHolder(itemView)
+    }
+
+    override fun onBindViewHolder(holder: MowSessionViewHolder, position: Int) {
+        val mowSession = mowSessions[position]
+        holder.date.text = getDateFromTimeStamp(mowSession.start.seconds)
+        holder.status.text = getMowingStatus(mowSession)
+        holder.collisions.text = context.getString(
+            R.string.avoided_collisions,
+            mowSession.avoidedCollisions.size.toString()
+        )
+        holder.itemView.setOnClickListener { onItemClicked(mowSession) }
+        if (mowSessionIsComplete(mowSession)) {
+            holder.mowingTime.text = context.getString(
+                R.string.mowing_time,
+                getMowingTimeMinutes(mowSession.start.seconds, mowSession.end!!.seconds)
+            )
+        } else {
+            holder.mowingTime.text = context.getString(
+                R.string.mowing_time,
+                getMowingTimeMinutes(
+                    mowSession.start.seconds,
+                    Instant.now().epochSecond
+                )
+            )
+        }
+    }
+
+    fun updateData(newData: List<MowSession>) {
+        mowSessions = newData
+        notifyDataSetChanged()
+    }
+
+    private fun getMowingStatus(mowSession: MowSession): String {
+        if (mowSessionIsComplete(mowSession)) {
+            return context.getString(R.string.mowing_status_complete)
+        }
+        return context.getString(R.string.mowing_status_active)
+    }
+
+    private fun mowSessionIsComplete(mowSession: MowSession): Boolean {
+        return mowSession.end != null
+    }
+
+    private fun getMowingTimeMinutes(startTimeStamp: Long, endTimeStamp: Long): String {
```

Comment:
If you feel like doing it, I think displaying the time in hh:mm could be pretty cool. But that is just me being nitpicky

Reply:
Sure, this can be a better way of showing it.